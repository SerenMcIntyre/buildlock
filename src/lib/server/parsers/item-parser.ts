// import fs from 'fs';g

export const parseItems = (filePath: string) => {
	return filePath;
	// const file = fs.readFileSync(filePath, 'utf8');
	// const items = file.split('\n');
	// const parsedItems = items.map((item) => {
	// 	const itemObject = JSON.parse(item);
	// 	return itemObject;
	// });
	// return parsedItems;
};
