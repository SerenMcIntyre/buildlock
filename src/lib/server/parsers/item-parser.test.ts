import { describe, it, expect } from 'vitest';
import { parseItems } from './item-parser';

describe('parseItems', () => {
	it('should parse items', () => {
		const filePath = './src/lib/server/parsers/example-item.vdata';
		const parsedItems = parseItems(filePath);
		console.debug(parsedItems);
		expect(parsedItems).toBeTruthy();
	});
});
