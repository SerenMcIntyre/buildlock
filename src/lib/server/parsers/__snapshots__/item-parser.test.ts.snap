// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseItems > should parse items 1`] = `
[
  {
    "abilityProperties": {
      "canSetTokenOverride": true,
      "cssClass": "range",
      "disableValue": "",
      "id": "upgrade_tech_damage_pulse",
      "itemId": "upgrade_tech_damage_pulse",
      "name": "0",
      "subclassScaleFunction": "{"_class":"scale_function_single_stat","_my_subclass_name":"AbilityCastRange_scale_function","m_eSpecificStatScaleType":"ETechRange"}",
      "value": "0",
    },
    "editor": {
      "folderName": "Tech",
      "id": "upgrade_tech_damage_pulse",
      "itemId": "upgrade_tech_damage_pulse",
    },
    "item": {
      "abilityActivation": "CITADEL_ABILITY_ACTIVATION_PASSIVE",
      "abilityTargetingLocation": "CITADEL_ABILITY_TARGETING_LOCATION_SELF",
      "abilityType": "EAbilityType_Item",
      "autoIntrinsicModifiers": "[{"_my_subclass_name":"tech_pulse_auto_intrinsic","_class":"modifier_base","m_bIsHidden":true,"m_vecAutoRegisterModifierValueFromAbilityPropertyName":["BonusHealth","TechPower"]}]",
      "bitsPostCastEnabledStateMask": "MODIFIER_STATE_BUSY_WITH_ACTION",
      "cancelAbilityKey": "+in_cancel_ability",
      "castAnimGraphParam": "e_GenericCast",
      "castAnimSequenceName": "ability_generic_cast",
      "class": "upgrade_tech_damage_pulse",
      "description": "",
      "disabled": false,
      "id": "upgrade_tech_damage_pulse",
      "image": "{images}/upgrades/mods_tech/torment_aura.psd",
      "itemClass": "upgrade_tech_damage_pulse",
      "itemSlotType": "EItemSlotType_Tech",
      "multibase": "["tech_upgrade_t3"]",
      "previewPathParticle": "",
      "price": 0,
      "rarity": "",
      "selectionNameOverride": "item",
      "shopFilters": "EShopFilterMagicDamage | EShopFilterDurability",
      "startTrained": true,
      "tier": "EModTier_3",
      "type": "EAbilityType_Item",
      "updateTime": 0,
    },
    "key": "upgrade_tech_damage_pulse",
    "tooltipSectionInfo": [
      {
        "abilitySectionType": "EArea_Innate",
        "id": "upgrade_tech_damage_pulse_0",
        "itemId": "upgrade_tech_damage_pulse",
        "sectionAttributes": "[{"m_vecAbilityProperties":["BonusHealth","TechPower"]}]",
      },
      {
        "abilitySectionType": "EArea_Passive",
        "id": "upgrade_tech_damage_pulse_1",
        "itemId": "upgrade_tech_damage_pulse",
        "sectionAttributes": "[{"m_strLocString":"#upgrade_tech_damage_pulse_desc","m_vecImportantAbilityProperties":[{"m_strImportantProperty":"DamagePulseAmount"}],"m_vecAbilityProperties":["DamagePulseRadius"]}]",
      },
    ],
  },
  {
    "abilityProperties": {
      "canSetTokenOverride": true,
      "cssClass": "range",
      "disableValue": "",
      "id": "upgrade_medic_bullets",
      "itemId": "upgrade_medic_bullets",
      "name": "0",
      "subclassScaleFunction": "{"_class":"scale_function_single_stat","_my_subclass_name":"AbilityCastRange_scale_function","m_eSpecificStatScaleType":"ETechRange"}",
      "value": "0",
    },
    "editor": {
      "folderName": "Weapon",
      "id": "upgrade_medic_bullets",
      "itemId": "upgrade_medic_bullets",
    },
    "item": {
      "abilityActivation": "CITADEL_ABILITY_ACTIVATION_PASSIVE",
      "abilityTargetingLocation": "CITADEL_ABILITY_TARGETING_LOCATION_SELF",
      "abilityType": "EAbilityType_Item",
      "autoIntrinsicModifiers": "[{"_class":"modifier_medic_bullets","_my_subclass_name":"modifier_explosive_bullets","m_bIsHidden":true,"m_ExplodeParticle":"particles/upgrades/crack_shot_hit.vpcf","m_ExplodeSound":"Upgrade.ExplosiveBullets.Explode","m_BuildUpModifier":{"_class":"","_my_subclass_name":""},"m_bRollOnceForAllBulletsInAShot":true,"m_bCanProcMultipleTimesFromSameShot":false,"m_bCanProcMultipleTimesOnOneTarget":false,"m_bShouldApplyAbilityCooldown":true,"m_TracerAdditionParticle":"particles/upgrades/medic_bullets_tracer.vpcf","m_bRequiresTargetFilter":true,"m_nAbilityTargetTypes":"CITADEL_UNIT_TARGET_ALL_ENEMY | CITADEL_UNIT_TARGET_GOLD_ORBS","m_ImpactParticle":"particles/upgrades/medic_bullets_tracer_impact.vpcf","m_ProcSound":"Player.Heal.Small"},{"_class":"modifier_regenerating_bullet_shield","_my_subclass_name":"modifier_regenerating_bullet_shield","m_bIsHidden":true,"m_vecAutoRegisterModifierValueFromAbilityPropertyName":["BaseAttackDamagePercent"]}]",
      "bitsPostCastEnabledStateMask": "MODIFIER_STATE_BUSY_WITH_ACTION",
      "cancelAbilityKey": "+in_cancel_ability",
      "castAnimGraphParam": "e_GenericCast",
      "castAnimSequenceName": "ability_generic_cast",
      "class": "citadel_item",
      "description": "",
      "disabled": false,
      "id": "upgrade_medic_bullets",
      "image": "{images}/upgrades/mods_weapon/medic_bullets.psd",
      "itemClass": "citadel_item",
      "itemSlotType": "EItemSlotType_WeaponMod",
      "multibase": "["weapon_upgrade_t1"]",
      "previewPathParticle": "",
      "price": 0,
      "rarity": "",
      "selectionNameOverride": "item",
      "shopFilters": "EShopFilterWeaponDamage | EShopFilterHealing",
      "startTrained": true,
      "tier": "EModTier_1",
      "type": "EAbilityType_Item",
      "updateTime": 1707443273,
    },
    "key": "upgrade_medic_bullets",
    "tooltipSectionInfo": [
      {
        "abilitySectionType": "EArea_Innate",
        "id": "upgrade_medic_bullets_0",
        "itemId": "upgrade_medic_bullets",
        "sectionAttributes": "[{"m_vecAbilityProperties":["BulletShieldMaxHealth","BaseAttackDamagePercent"]}]",
      },
      {
        "abilitySectionType": "",
        "id": "upgrade_medic_bullets_1",
        "itemId": "upgrade_medic_bullets",
        "sectionAttributes": "[{"m_vecAbilityProperties":["AbilityCooldown"],"m_vecImportantAbilityProperties":[{"m_strImportantProperty":"HealFromHero"},{"m_strImportantProperty":"HealFromNPC"}],"m_strLocString":"#upgrade_medic_bullets_desc"}]",
      },
    ],
  },
  {
    "abilityProperties": {
      "canSetTokenOverride": true,
      "cssClass": "range",
      "disableValue": "",
      "id": "weapon_alternative_rmb_lmb_activate",
      "itemId": "weapon_alternative_rmb_lmb_activate",
      "name": "0",
      "subclassScaleFunction": "{"_class":"scale_function_single_stat","_my_subclass_name":"AbilityCastRange_scale_function","m_eSpecificStatScaleType":"ETechRange"}",
      "value": "0",
    },
    "editor": {
      "folderName": "Base",
      "id": "weapon_alternative_rmb_lmb_activate",
      "itemId": "weapon_alternative_rmb_lmb_activate",
    },
    "item": {
      "abilityActivation": "CITADEL_ABILITY_ACTIVATION_ON_BUTTON_IS_DOWN",
      "abilityTargetingLocation": "CITADEL_ABILITY_TARGETING_LOCATION_SELF",
      "abilityType": "EAbilityType_Weapon",
      "autoIntrinsicModifiers": "[]",
      "bitsPostCastEnabledStateMask": "MODIFIER_STATE_BUSY_WITH_ACTION",
      "cancelAbilityKey": "+in_cancel_ability",
      "castAnimGraphParam": "",
      "castAnimSequenceName": "",
      "class": "citadel_base_ability",
      "description": "",
      "disabled": false,
      "id": "weapon_alternative_rmb_lmb_activate",
      "image": "{images}/hud/abilities/weapon_damage.psd",
      "itemClass": "citadel_base_ability",
      "itemSlotType": "",
      "multibase": "["common_properties","weapon_info_base"]",
      "previewPathParticle": "",
      "price": 0,
      "rarity": "",
      "selectionNameOverride": "",
      "shopFilters": "",
      "startTrained": true,
      "tier": "",
      "type": "EAbilityType_Weapon",
      "updateTime": 0,
    },
    "key": "weapon_alternative_rmb_lmb_activate",
    "tooltipSectionInfo": [],
  },
]
`;
